float64 pos
float64 speed
float64 force
duration time_from_start